faunadb-scala/src/main/scala/faunadb/values/Time.scala

```diff
@@ -0,0 +1,102 @@
+package faunadb.values
+package time
+
+import org.joda.time.Instant
+import org.joda.time.format.ISODateTimeFormat
+import scala.Ordered._
+
+/**
+  * Represents the instant from Java's epoch composed by the number of seconds since eposh
+  * and the nanoseconds offset.
+  */
+class HighPrecisionTime private(val secondsSinceEpoch: Long, val nanoSecondsOffset: Int) {
```

Comment:
other style nit: Seconds should not be capitalized when part of method names as part of milliseconds, microseconds, or nanoseconds, since those are all single words.